test/test_ad.jl

```diff
@@ -39,6 +50,17 @@ forwarddiff_color_jacobian!(_J1, f, x, colorvec = repeat(1:3,10))
 @test fcalls == 1
 
 fcalls = 0
+_J1 = forwarddiff_color_jacobian(oopf, x, colorvec = repeat(1:3,10), sparsity = _J, jac_prototype = _J)
```

Comment:
we should test with Tracker.TrackedArray and StaticArrays. I think these will currently fail.

Reply:
Yes, but I have no idea about writing a real one.

Reply:
I made one, but I would expect a really bad performance. @ChrisRackauckas